### PR TITLE
Add mailTo as config to apply button

### PR DESCRIPTION
**Pull request by abnersajr**
#### What does this PR do?

Solve issue #31.
Adds `mailTo` key as theme config.

#### Showroom

image

#### Where should the reviewer start?

From the key names and Button component inside `Job` component

#### How should this be manually tested?

Get an example job, configure your theme with a valid `mailTo` value. Go to a Job page check if the button is there.
Bad path can be tested leaving the `mailTo` key unfilled on config.

#### Any background context you want to provide?

It uses site metadata hook to get value from config.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -34,23 +34,30 @@ module.exports = {
...
+| `spaceId`                | `string` | Your Contentful space ID                                   | `true`   |
+| `accessToken`            | `string` | Your Contentful content delivery API access token          | `true`   |
+| `managementToken`        | `string` | Your Contentful personal access                            | `true`   |
+| `environmentId`          | `string` | Your Contentful environment ID. (e.g. `master`)            | `true`   |
+| `mailTo`                 | `string` | The e-mail address for people apply                        | `true`   |
+| `googleAnalyticsOptions` | `object` | Your Google Analytics options                              | `false`  |
 
 Create your Contentful space, generate all the needed api keys and fill them in.
 
+📊 :Google Analytics is optional, however by default this is placed on body and you **must** pass the key `trackingId` to have the GA script included.
```
Comment (matheusazzi) [inline README.md]: @abnersajr That change seems unrelated to current PR. I think it's mixed up with the `add-analytics-support` branch

Comment (abnersajr) [reply README.md]: I will fix it.

File: packages/gatsby-theme-careers/src/components/Job.js [hunk 2]
```diff
@@ -4,23 +4,33 @@ import { Box, Heading } from 'flokit'
...
+      </Box>
 
-      <Button marginTop='5'>Apply</Button>
-    </article>
-  </Layout>
-)
+      <article>
+        {job.body && documentToReactComponents(job.body.json)}
+
+        {mailTo && (
```
Comment (matheusazzi) [inline packages/gatsby-theme-careers/src/components/Job.js]: Maybe we could make it an apply component that receives an email address and subject. But I don't have any strong opinion regarding that.

Comment (abnersajr) [reply packages/gatsby-theme-careers/src/components/Job.js]: This is beign used only here and that's why didn't extract it. I don't feel the necessity because I don't see this beign used in other places at least for now. 🤔

Comment (abnersajr) [reply packages/gatsby-theme-careers/src/components/Job.js]: Extracted to `ApplyButton` component.

File: README.md [hunk 3]
```diff
@@ -34,20 +34,22 @@ module.exports = {
         accessToken: '',
         managementToken: '',
         environmentId: '',
+        mailTo: 'become@jobs.com'
```
Comment (iagodahlem) [inline README.md]: Can you move this to be at first in the object?

File: README.md [hunk 4]
```diff
@@ -34,20 +34,22 @@ module.exports = {
...
-| `environmentId`   | `string` | Your Contentful environment ID. (e.g. `master`)            | `true`   |
+| Option Name              | Type     | Description                                                | Required |
+| :----------------------- | :------- | :--------------------------------------------------------- | :------- |
+| `basePath`               | `string` | The base path where your site will live. (e.g. `/careers`) | `false`  |
+| `title`                  | `string` | The main title used in the header                          | `false`  |
+| `spaceId`                | `string` | Your Contentful space ID                                   | `true`   |
+| `accessToken`            | `string` | Your Contentful content delivery API access token          | `true`   |
+| `managementToken`        | `string` | Your Contentful personal access                            | `true`   |
+| `environmentId`          | `string` | Your Contentful environment ID. (e.g. `master`)            | `true`   |
+| `mailTo`                 | `string` | The e-mail address for people apply                        | `true`   |
```
Comment (iagodahlem) [inline README.md]: Can you move this to between `title` and `spaceId`?

File: packages/example/gatsby-config.js [hunk 5]
```diff
@@ -10,6 +10,7 @@ module.exports = {
         managementToken: process.env.CONTENTFUL_CONTENT_MANAGEMENT_TOKEN,
         environmentId: process.env.CONTENTFUL_ENVIRONMENT_ID,
         downloadLocal: true,
+        mailTo: 'become@jobs.com'
```
Comment (iagodahlem) [inline packages/example/gatsby-config.js]: Same thing, let's move it to be at first in the object.

File: packages/gatsby-theme-careers/src/components/ApplyButton.js [hunk 6]
```diff
@@ -0,0 +1,17 @@
+import React from 'react'
+import Button from './Button'
+
+const ApplyButton = ({ mailTo, subject, ...restProps }) => {
```
Comment (iagodahlem) [inline packages/gatsby-theme-careers/src/components/ApplyButton.js]: I think you can just use `...props` for the rest of the props.

Comment (iagodahlem) [reply packages/gatsby-theme-careers/src/components/ApplyButton.js]: Do you think we need a specific `subject` prop? What do you think about just passing it together with the `mailTo` prop?

Comment (abnersajr) [reply packages/gatsby-theme-careers/src/components/ApplyButton.js]: Fixed. Removed the subject and the `restProps` to `props` 👍
